tests/ibc_pfm_test.go

```diff
@@ -0,0 +1,351 @@
+	dymensionUser, rollappUser, osmosisUser := users[0], users[1], users[2]
+
+	dymensionUserAddr := dymensionUser.FormattedAddress()
+	rollappUserAddr := rollappUser.FormattedAddress()
+	osmosisUserAddr := osmosisUser.FormattedAddress()
+
+	// Get original account balances
+	dymensionOrigBal, err := dymension.GetBalance(ctx, dymensionUserAddr, dymension.Config().Denom)
+	require.NoError(t, err)
+	require.Equal(t, walletAmount, dymensionOrigBal)
+
+	rollappOrigBal, err := rollapp1.GetBalance(ctx, rollappUserAddr, rollapp1.Config().Denom)
+	require.NoError(t, err)
+	require.Equal(t, walletAmount, rollappOrigBal)
+
+	osmosisOrigBal, err := osmosis.GetBalance(ctx, osmosisUserAddr, osmosis.Config().Denom)
+	require.NoError(t, err)
+	require.Equal(t, walletAmount, osmosisOrigBal)
+
+	t.Run("multihop rollapp->dym->osmosis", func(t *testing.T) {
+		firstHopDenom := transfertypes.GetPrefixedDenom(channDymRollApp.PortID, channDymRollApp.ChannelID, rollapp1.Config().Denom)
+		secondHopDenom := transfertypes.GetPrefixedDenom(channOsmosDym.PortID, channOsmosDym.ChannelID, firstHopDenom)
+
+		firstHopDenomTrace := transfertypes.ParseDenomTrace(firstHopDenom)
+		secondHopDenomTrace := transfertypes.ParseDenomTrace(secondHopDenom)
+
+		firstHopIBCDenom := firstHopDenomTrace.IBCDenom()
+		secondHopIBCDenom := secondHopDenomTrace.IBCDenom()
+
+		fmt.Println(firstHopDenom)
+		fmt.Println(firstHopDenomTrace)
+		fmt.Println(firstHopIBCDenom)
+
+		fmt.Println(secondHopDenom)
+		fmt.Println(secondHopDenomTrace)
+		fmt.Println(secondHopIBCDenom)
+
+		zeroBal := math.ZeroInt()
+		transferAmount := math.NewInt(100_000)
+
+		// Send packet from rollapp1 -> dym -> osmosis
+		transfer := ibc.WalletData{
+			Address: dymensionUserAddr,
+			Denom:   rollapp1.Config().Denom,
+			Amount:  transferAmount,
+		}
+
+		firstHopMetadata := &PacketMetadata{
+			Forward: &ForwardMetadata{
+				Receiver: osmosisUserAddr,
+				Channel:  channDymOsmos.ChannelID,
+				Port:     channDymOsmos.PortID,
+				Timeout:  5 * time.Minute,
+			},
+		}
+
+		memo, err := json.Marshal(firstHopMetadata)
+		require.NoError(t, err)
+
+		transferTx, err := rollapp1.SendIBCTransfer(ctx, channsRollAppDym.ChannelID, rollappUser.KeyName(), transfer, ibc.TransferOptions{Memo: string(memo)})
+		require.NoError(t, err)
+		err = transferTx.Validate()
+		require.NoError(t, err)
+
+		err = testutil.WaitForBlocks(ctx, 50, rollapp1)
+		require.NoError(t, err)
+
+		rollAppBalance, err := rollapp1.GetBalance(ctx, rollappUserAddr, rollapp1.Config().Denom)
+		require.NoError(t, err)
+
+		dymBalance, err := dymension.GetBalance(ctx, dymensionUserAddr, firstHopIBCDenom)
+		require.NoError(t, err)
+
+		osmosisBalance, err := osmosis.GetBalance(ctx, osmosisUserAddr, secondHopIBCDenom)
+		require.NoError(t, err)
+
+		require.True(t, rollAppBalance.Equal(walletAmount.Sub(transferAmount)))
+		require.True(t, dymBalance.Equal(zeroBal))
+		require.True(t, osmosisBalance.Equal(transferAmount))
```

Comment:
We could use AssertBalance method instead to be consistent with other tests.

Reply:
Also, when it fails, message is misleading. If you want to keep using `require.True`, then consider adding error message, like
```
equire.True(t, osmosisBalance.Equal(transferAmount), "error message in case of mismatch")
```